dev/DynamicDependency/MddBootstrap.cpp

```diff
@@ -0,0 +1,378 @@
+    // in the Dynamic Dependency Lifetime Manager package's dependencies,
+    // as resolved by Windows. A user can have multiple framework packages
+    // in a family registered at a time, for multiple reasons:
+    //
+    //   * Multiple Architectures -- x86/x64 on an x64 machine, x86/arm/arm64/x86ona64 on an arm64 machine, etc
+    //   * Multiple Versions -- v1.0.0.0 in use by processes running as pkg1 and v1.0.0.1 in use by processes running as pkg2
+    //                          or v1.0.0.0 in use by running processes and v1.0.0.1 in package graphs for packages w/no running process
+    //
+    // Thus FindPackagesByPackageFamily(pkgfamilyname,...) and PackageManager.FindPackages(user="", pkgfamilyname) could be ambiguous.
+    // We need the actual dependency graph known to Windows for the DDLM package where we got our LifetimeManager.
+    // That leaves us few options:
+    //
+    //   * PackageManager.FindPackage(user="", lifetimeManager->GetPackageFullName()).Dependencies
+    //   * GetPackageInfo(OpenPackageInfoByFullName(lifetimeManager->GetPackageFullName())
+    //
+    // We'll go with the latter as the simpler (no COM/WinRT) and more performant solution.
+
+    // Fetch the package graph for the package (per packageFullName)
+    wil::unique_package_info_reference packageInfoReference;
+    THROW_IF_WIN32_ERROR(OpenPackageInfoByFullName(packageFullName, 0, &packageInfoReference));
+    UINT32 bufferLength{};
+    UINT32 packageInfoCount{};
+    const auto hr{ HRESULT_FROM_WIN32(GetPackageInfo(packageInfoReference.get(), PACKAGE_FILTER_DIRECT, &bufferLength, nullptr, &packageInfoCount)) };
+    THROW_HR_IF(hr, hr != HRESULT_FROM_WIN32(ERROR_INSUFFICIENT_BUFFER));
+    THROW_HR_IF(E_UNEXPECTED, packageInfoCount == 0);
+    auto buffer{ wil::make_unique_cotaskmem<BYTE[]>(bufferLength) };
+    THROW_IF_WIN32_ERROR(GetPackageInfo(packageInfoReference.get(), PACKAGE_FILTER_DIRECT, &bufferLength, buffer.get(), &packageInfoCount));
+
+    // Find the Project Reunion Framework package in the package graph to determine its path
+    const PACKAGE_INFO* packageInfo{ reinterpret_cast<const PACKAGE_INFO*>(buffer.get()) };
+    for (size_t index = 0; index < packageInfoCount; ++index, ++packageInfo)
+    {
+        PCWSTR frameworkPackageFamilyName{ L"Microsoft.ProjectReunion.Framework_8wekyb3d8bbwe" };
+        if (CompareStringOrdinal(packageInfo->packageFamilyName, -1, frameworkPackageFamilyName, -1, TRUE) == CSTR_EQUAL)
+        {
+            frameworkPackageInfo = packageInfo;
+            return buffer;
+        }
+    }
+
+    // We didn't find the framework package as a dependency of the package
+    // providing the LifetimeManager.
+    //
+    // Verify the package providing the LifetimeManager declares
+    // <PackageDependency> on the Project Reunion Framework package.
+    THROW_WIN32(APPMODEL_ERROR_PACKAGE_NOT_AVAILABLE);
+}
+
+DLL_DIRECTORY_COOKIE AddFrameworkToPath(PCWSTR frameworkPath)
```

Comment:
Just have this return `wil::unique_dll_directory_cookie` directly. Avoids you releasing here and re-attaching on the other side of the call.